File: app/src/main/java/com/example/foreground/util/NetworkHelper.java
package com.example.foreground.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;


public class NetworkHelper {


    public static boolean isInternetAvailable(Context pContext) {
        if (pContext == null) {
            return false;
        }
        ConnectivityManager cm =
                (ConnectivityManager) pContext.getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();

        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting();
    }

}

File: appK/src/main/java/com/example/foreground/util/NetworkHelper.kt
package com.example.foreground.util

import android.content.Context
import android.net.ConnectivityManager

object NetworkHelper {
    @JvmStatic
    fun isInternetAvailable(pContext: Context?): Boolean {
        if (pContext == null) {
            return false
        }
        val cm = pContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = cm.activeNetworkInfo
        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting
    }
}